urllib3/util/url.py

```diff
@@ -192,7 +192,10 @@ def parse_url(url):
             # If given, ports must be integers.
             if not port.isdigit():
                 raise LocationParseError(url)
-            port = int(port)
+            try:
```

Comment:
Actually, I've changed my mind here. I agree ports need to be integers, so the `isdigit` check remains valuable.